### PR TITLE
Creates an nd4j (assembly|uber)-jar, tags the build with the git-commit-id plugin.

### PR DESCRIPTION
**Pull request by huitseeker**
Baisc changes towards an nd4j (rpm|deb) package.

### CODE REVIEW

File: nd4j-uberjar/pom.xml [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+    <parent>
+        <artifactId>nd4j</artifactId>
+        <groupId>org.nd4j</groupId>
+        <version>0.8.1-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>nd4j-uberjar</artifactId>
```
Comment (agibsonccc) [inline nd4j-uberjar/pom.xml]: Are we going to be able to skip this module when uploading to maven central?

File: nd4j-uberjar/pom.xml [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+      </dependency>
+      <dependency>
+        <groupId>org.nd4j</groupId>
+        <artifactId>nd4j-gson</artifactId>
+        <version>${project.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.nd4j</groupId>
+        <artifactId>nd4j-jackson-reflectionloader</artifactId>
+        <version>${project.version}</version>
```
Comment (agibsonccc) [inline nd4j-uberjar/pom.xml]: I wonder about dependnecy conflicts with the serialization libraries..

File: nd4j-uberjar/pom.xml [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        <artifactId>nd4j-native</artifactId>
+        <version>${project.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.nd4j</groupId>
+        <artifactId>nd4j-native-platform</artifactId>
+        <version>${project.version}</version>
+      </dependency>
+      <dependency>
+        <groupId>org.nd4j</groupId>
```
Comment (agibsonccc) [inline nd4j-uberjar/pom.xml]: This looks misguided. Both backends being on the classpath? How will that even work?

File: nd4j-uberjar/pom.xml (hunk 1 above)
Comment (huitseeker) [inline nd4j-uberjar/pom.xml]: Now, without `-Puberjar`, it's a 2.7KB jar.

File: nd4j-uberjar/pom.xml (hunk 3 above)
Comment (huitseeker) [inline nd4j-uberjar/pom.xml]: You now have to pass `-Pcuda` or `-Pnative` along with `-Puberjar` to build one of the uberjars.

File: nd4j-uberjar/pom.xml (hunk 2 above)
Comment (huitseeker) [inline nd4j-uberjar/pom.xml]: How would we address that ?

**Pull request merged**